admin/src/templates/edit_user.html

```diff
@@ -64,13 +65,38 @@
       <span class="help-block">{{errors.contact}}</span>
     </div>
 
-    <div class="form-group" ng-class="{'has-error': errors.password, 'required': !editUserModel.id}">
+    <div class="form-group" ng-if="allowTokenLogin && !editUserModel.tokenLoginEnabled">
+      <input type="checkbox" id="tokenLogin" ng-true-value="true" ng-false-value="" ng-model="editUserModel.token_login">
+      <label for="tokenLogin" translate class="form-check-label">configuration.enable.token.login</label>
+      <div class="help-block" translate>configuration.enable.token.login.help</div>
+    </div>
+
+    <div class="form-group" ng-if="allowTokenLogin && editUserModel.tokenLoginEnabled">
+      <p>
+        <label class="form-check-label">
+          <span ng-if="editUserModel.tokenLoginEnabled.active && !editUserModel.tokenLoginEnabled.expired" translate translate-values="{ date: '{{editUserModel.tokenLoginEnabled.expirationDate}}' }">configuration.enable.token.login.enabled.active</span>
+          <span ng-if="editUserModel.tokenLoginEnabled.active && editUserModel.tokenLoginEnabled.expired" translate translate-values="{ date: '{{editUserModel.tokenLoginEnabled.expirationDate}}' }">configuration.enable.token.login.enabled.expired</span>
+          <span ng-if="!editUserModel.tokenLoginEnabled.active" translate translate-values="{ date: '{{editUserModel.tokenLoginEnabled.loginDate}}' }">configuration.enable.token.login.enabled.inactive</span>
+        </label>
+      </p>
+      <div>
+        <input type="checkbox" id="disabletokenLogin" ng-true-value="false" ng-false-value="" ng-model="editUserModel.token_login">
+        <label for="disabletokenLogin" translate class="form-check-label">configuration.enable.token.login.disable</label>
+      </div>
+      <div>
+        <input type="checkbox" id="refreshTokenLogin" ng-true-value="true" ng-false-value="" ng-model="editUserModel.token_login">
+        <label for="refreshTokenLogin" translate class="form-check-label">configuration.enable.token.login.refresh</label>
+      </div>
+      <div class="help-block" translate>configuration.enable.token.login.refresh.help</div>
+    </div>
+
+    <div class="form-group" ng-class="{ 'has-error': errors.password, 'required': !editUserModel.id, 'hidden': editUserModel.token_login || (editUserModel.token_login !== false && editUserModel.tokenLoginEnabled) }">
```

Comment:
No token:
![image](https://user-images.githubusercontent.com/35681649/86962418-8a0d2480-c16b-11ea-9279-41f12681fdf1.png)

Enabling token:
![image](https://user-images.githubusercontent.com/35681649/86962435-942f2300-c16b-11ea-9b0d-cc4943313795.png)

Token enabled: 
![image](https://user-images.githubusercontent.com/35681649/86962488-b1fc8800-c16b-11ea-9c66-b6fd5441f4de.png)

Token enabled, choosing to disable: 
![image](https://user-images.githubusercontent.com/35681649/86962523-bcb71d00-c16b-11ea-9e2a-6e2599b44f57.png)

Token enabled choosing to regenerate: 
![image](https://user-images.githubusercontent.com/35681649/86962564-cccefc80-c16b-11ea-9c6d-a562e548039b.png)

Link expired:
![image](https://user-images.githubusercontent.com/35681649/86962813-28998580-c16c-11ea-8979-a2dc0c0e5678.png)

Link accessed: 
![image](https://user-images.githubusercontent.com/35681649/86962756-14558880-c16c-11ea-8138-444d112203be.png)

Reply:
These options should be radio buttons because it doesn't make sense to pick them both, which means we also need a "do nothing" option selected by default.

It's a complicated UI that probably needs to be redone at some point (eg: so it's not a modal!) but I think this is good enough for now.

Reply:
They "act" as radio buttons already because they're updating the same value. Both left unchecked means the value remains "undefined". 
I can switch to radios if you think it's clearer tho. 
Agree that it will have to change, I don't like it much myself!

Reply:
Right but it's surprising to the user if they expect them to act like checks. I'd prefer radios, but it's not a deal breaker.

Reply:
Cool, will do!